src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCacheFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers
+{
+    public class EventHubQueueCacheFactory : IEventHubQueueCacheFactory
+    {
+        private readonly EventHubStreamProviderSettings _providerSettings;
+        private readonly SerializationManager _serializationManager;
+        private readonly FixedSizeObjectPool<FixedSizeBuffer> _bufferPool;
+        private readonly TimePurgePredicate _timePurge;
+
+        public EventHubQueueCacheFactory(EventHubStreamProviderSettings providerSettings, SerializationManager serializationManager)
+        {
+            _providerSettings = providerSettings;
+            _serializationManager = serializationManager;
+
+            _bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(_providerSettings.CacheSizeMb, () => new FixedSizeBuffer(1 << 20));
```

Comment:
Since currently it is created in ctor, a virtual method won't really work. Instead, I've added an alternative ctor allowing to pass a custom pool.